src/options.jl

```diff
@@ -157,9 +155,26 @@ function print_ignored_options(logger,option_dict)
     end
 end
 
+function get_primary_options(options)
```

Comment:
I would maybe rename this function `_get_primary_options` to emphasize that it is internal.